src/utils/vue/script-setup.ts

```diff
@@ -0,0 +1,365 @@
+        `${DEFINE_PROPS} declaration.`,
+      node,
+    );
+  }
+  if (!context.ctx.propsRuntimeDefaults) {
+    context.ctx.error(
+      `The 2nd argument of ${WITH_DEFAULTS} is required.`,
+      node,
+    );
+  }
+
+  return res;
+}
+function processDefineModel(
+  node: Expression,
+  context: Context,
+): string | undefined {
+  if (!isCallOf(node, DEFINE_MODEL)) {
+    return;
+  }
+
+  const [modelNameDecl, modelRuntimeDecl] = getDefineModelRuntimeDecl(
+    node,
+    context,
+  );
+
+  const modelTypeDecl = node.typeParameters?.params[0];
+  if (!modelTypeDecl) {
+    return;
+  }
+
+  const model = context.utils.inferRuntimeType(context.ctx, modelTypeDecl);
+  if (!model || model.length === 0 || model[0] === "Unknown") {
+    return;
+  }
+
+  // { type: String } or { type: [String, Number] } if model have multiple types
+  const modelCodegenTypeDecl =
+    model.length === 1
+      ? ({ type: "Identifier", name: model[0] } satisfies Identifier)
+      : ({
+          type: "ArrayExpression",
+          elements: model.map(
+            (name) => ({ type: "Identifier", name }) satisfies Identifier,
+          ),
+        } satisfies ArrayExpression);
+
+  const modelCodegenDecl: ObjectExpression = {
+    type: "ObjectExpression",
+    properties: [
+      {
+        type: "ObjectProperty",
+        key: { type: "StringLiteral", value: "type" },
+        value: modelCodegenTypeDecl,
+        computed: false,
+        shorthand: false,
+      } satisfies ObjectProperty,
+    ],
+  };
+  if (modelRuntimeDecl) {
+    modelCodegenDecl.properties.push({
+      type: "SpreadElement",
+      argument: modelRuntimeDecl,
+    });
+  }
+
+  node.typeArguments = undefined;
+  node.typeParameters = undefined;
+  node.arguments = modelNameDecl
+    ? [modelNameDecl, modelCodegenDecl]
+    : [modelCodegenDecl];
+
+  return context.utils.babel.generate(node).code;
+}
+
+function getDefineModelRuntimeDecl(
+  node: CallExpression,
+  context: Context,
+): [StringLiteral | undefined, ObjectExpression | undefined] {
+  const [arg0, arg1] = node.arguments;
+  if (arg0 && arg0.type === "StringLiteral") {
+    if (arg1 && arg1.type !== "ObjectExpression") {
+      context.ctx.error(
+        `${DEFINE_MODEL}()'s second argument must be an object.`,
+        arg1,
+      );
+    }
+
+    return [arg0, arg1 as ObjectExpression];
+  }
+
+  if (arg0 && arg0.type !== "ObjectExpression") {
+    context.ctx.error(`Unexpected argument type for ${DEFINE_MODEL}().`, arg0);
+  }
+
+  return [undefined, arg0 as ObjectExpression | undefined];
+}
+
+async function prepareContext(
+  { script, scriptSetup }: SFCDescriptor,
+  id: string,
+): Promise<Context> {
+  const {
+    extractRuntimeProps,
+    extractRuntimeEmits,
+    inferRuntimeType,
+    MagicString,
+  } = await import("vue/compiler-sfc");
+  const { parse } = await import("@babel/parser");
+  const generate = (await import("@babel/generator").then(
+    (m) => m.default || m,
```

Comment:
```suggestion
    (m) => m.default?.generate || m.default || m,
```

Reply:
I think we can use normal import because it is dependency.
Mark it as peer dependency can reduce ~500k disk space for user who don't use vue loader, but I'm afraid most people will forget to install this.